### ISSUE TITLE
Full compatibility with go_amino?

### ISSUE THREAD

**Issue by adrianbrink** [tendermint/amino_rs#19]
Is `amino_rs` fully compatible with `go-amino`?

**Comment by Liamsi** [comment]
No, not yet. E.g., https://github.com/tendermint/amino_rs/issues/15 is missing. And the json encoding isn't part of this implementation at all (only binary encoding).

Also, it does allow everything else that protobuf does (e.g. `oneof`, encoding floats, maps are not allowed by default in go-amino but they are here).

**Comment by Liamsi** [comment]
Out of curiosity though: Do you have any use-case that you can't implement with what is there already?

**Comment by yihuang** [comment]
IMO, one of the incompatible issues is the handling of default/empty value, which caused some annoyments in tendermint-rs.
I've described them here: https://github.com/interchainio/tendermint-rs/pull/103#issuecomment-570100348 .
